warehouse/search/tasks.py

```diff
@@ -87,6 +87,9 @@ def _project_docs(db):
         .order_by(Release.name)
     )
 
+    if project_name:
+        release_data = release_data.filter(Release.name == project_name)
```

Comment:
Probably want to add this filter earlier on `releases_list` instead of here? Maybe it makes no difference...

Reply:
_smart_

Reply:
it's damn fast either way, but you're right!

Reply:
so, since `releases_list` is a subquery, it seems it is not easily modified after the fact. probably best to let this sleeping dog lay.

Reply:
You could just `.subquery` it later, and do `releases_list = releases_list.filter(...)` earlier. Again, might not really matter.

Reply:
easy enough.